### ISSUE TITLE
aspnet-user-identity is blank when using Azure AD SSO

### ISSUE THREAD

**Issue by MattKalafut** [NLog/NLog.Web#593]
**Type:** Question
**NLog Version:**
```xml
    <PackageReference Include="NLog" Version="4.7.2" />
    <PackageReference Include="NLog.Extensions.Logging" Version="1.6.4" />
    <PackageReference Include="NLog.Web.AspNetCore" Version="4.9.2" />
```
**Platform:** ASP.NET Core 2.2
**Current NLog Config**
```xml
<?xml version="1.0" encoding="utf-8" ?>
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      autoReload="true"
      throwConfigExceptions="true"
      internalLogLevel="Trace"
      internalLogFile="c:\temp\internal-nlog.txt">

  <!-- enable asp.net core layout renderers -->
  <extensions>
    <add assembly="NLog.Extensions.Logging"/>
    <add assembly="NLog.Web.AspNetCore"/>
  </extensions>

  <!-- the targets to write to -->
  <targets>
    <!-- write logs to file  -->
    <target xsi:type="File" name="allfile" fileName="c:\temp\nlog-all-${shortdate}.log"
            layout="${longdate}|${event-properties:item=EventId_Id}|${uppercase:${level}}|${logger}|${message} ${exception:format=tostring}" />

    <!-- another file log, only own logs. Uses some ASP.NET core renderers -->
    <target xsi:type="File" name="ownFile-web" fileName="c:\temp\nlog-own-${shortdate}.log"
            layout="${longdate}|${event-properties:item=EventId_Id}|${uppercase:${level}}|${logger}|${message} ${exception:format=tostring}|url: ${aspnet-request-url}|action: ${aspnet-mvc-action}" />

    <target name="databaseLogger" xsi:type="Database"
          dbProvider="sqlserver"
          dbHost="${configsetting:AppSettings.NlogConnection.DbHost}"
          dbDatabase="${configsetting:AppSettings.NlogConnection.Database}"
          dbUserName="${configsetting:AppSettings.NlogConnection.User}"
          dbPassword="${configsetting:AppSettings.NlogConnection.Password}">

      <commandText>
        insert into dbo.Log (
        CorrelationId, Application, UserName, Logged, Level, Message,
        Logger, CallSite, Exception
        ) values (
        @CorrelationId, @Application, @UserName, @Logged, @Level, @Message,
        @Logger, @Callsite, @Exception
        );
      </commandText>
      <parameter name="@correlationId" layout="${aspnet-traceidentifier}" />
      <parameter name="@application" layout="${configsetting:AppSettings.NlogConnection.AppName}" />
      <parameter name="@UserName" layout="${aspnet-user-identity}" />
      <parameter name="@logged" layout="${date}" />
      <parameter name="@level" layout="${level}" />
      <parameter name="@message" layout="${message}" />
      <parameter name="@logger" layout="${logger}" />
      <parameter name="@callSite" layout="${callsite:filename=true}" />
      <parameter name="@exception" layout="${exception:tostring}" />
    </target>

  </targets>

  <!-- rules to map from logger name to target -->
  <rules>
    <!--Skip non-critical Microsoft logs and so log only own logs-->
    <logger name="Microsoft.*" levels="Warn, Error, Fatal" writeTo="databaseLogger"></logger>
    <logger name="Microsoft.*" minLevel="Trace" final="true" />

    <!--All logs, including from Microsoft unless suppressed (which I do, above)-->
    <logger name="*" minlevel="Trace" writeTo="allfile" />
    <logger name="*" minlevel="${configsetting:AppSettings.NlogConnection.LogLevel" writeTo="databaseLogger" />
    <logger name="*" minlevel="Trace" writeTo="ownFile-web" />
  </rules>
</nlog>
```

I'm using Azure AD SSO as a login system for my application and after some troubleshooting I finally got the above configuration to work and write logging messages to the database.  The UserName field however is always blank in my log statements.  The user claim we're using to display on the application's front end is the user's email address.  Is there a way to configure the ```aspnet-user-identity``` Layout Renderer to grab the email once the user's logged in and send to the database when my log statements are written?

Would be happy to provide any additional info.  Thank you.

**Comment by MattKalafut** [comment]
@304NotModified Hey Julian, I am checking into that now.  I have my internal log at Debug now (It was set to Trace before while debugging this issue) and no errors show up at all and the message logs correctly to the DB just with a blank UserName field.

I will check and update shortly on the HTTPContextAccessor inquiry.

Thank you.

**Comment by MattKalafut** [comment]
@304NotModified During my debugging I noticed that in my first controller action I write a logging statement, I see the following inside of the HttpContext.User.Identity

![image](https://user-images.githubusercontent.com/4413605/88849470-8ccdc900-d1af-11ea-96df-431ac6adc44b.png)

But within the ```claims``` I do see that I have claims that I would be able to use to write out to the log to satisfy a "UserName" criteria.  It seems to be a mix of AD and other claims made from within Azure.

**Comment by MattKalafut** [comment]
@304NotModified I think the problem was on our end.  I talked to the developer who implemented the Azure SSO Login system and after some Googling we found [this](https://stackoverflow.com/questions/54444747/user-identity-name-is-null-after-federated-azure-ad-login-with-aspnetcore-2-2) StackOverflow question which was basically the same issue we had on our end.  And the corresponding answer fixed the issue and now I see the name within the logged messages in the database.

I guess as a quick follow-up, is there a way to bind this setting from a claim property rather than the specific User.Identity.Name field?

Thanks so much for your help and leading me down the path to find the fix!
